Plugins/SWIG/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+function(add_swig_plugin target interopfile)
+    add_custom_target(
+        Generate_${target}
+        COMMAND chmod a+x ./generate.sh && mkdir -p ./${target} && ${CMAKE_CURRENT_SOURCE_DIR}/generate.sh -v -c++ -fcompact -o ${interopfile} -outdir ./${target} -I../../NWNXLib/API ${ARGN}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    )
+
+    add_library(${target} SHARED ${interopfile})
+    set_property(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/${interopfile} PROPERTY GENERATED 1)
+    add_sanitizers(${target})
+    target_link_libraries(${target} Core "-Wl,--no-undefined")
+    set_target_properties(${target} PROPERTIES PREFIX "${PLUGIN_PREFIX}")
+    target_include_directories(${target} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+    target_compile_definitions(${target} PRIVATE "-DPLUGIN_NAME=\"${PLUGIN_PREFIX}${target}\"")
+    add_dependencies(${target} Generate_${target})
+endfunction()
+
+add_swig_plugin(SWIG_DotNET "DotNETInterop.cpp" -csharp -dllimport "${PLUGIN_PREFIX}SWIG_DotNET" -namespace "NWN.Native.API" "DotNET.i")
```

Comment:
If we are going to keep generating it every build, rather than do manually, we need two more things:
1. Skip the plugin altogether if people don't have the swig dependency installed. Adding a new plugin should never break people's builds.
2. Set up make dependencies on the API, so that it is only rebuilds the generated files if the source API files changed. If I'm reading this right, it will generate DotNETInterop.cpp every time I edit an unrelated file in a different plugin and try to build it.